src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/VepTranscriptFmsDTOValidator.java

```diff
@@ -0,0 +1,331 @@
+			if (cdsStartEnd == null) {
+				response.addErrorMessage("cdsPosition", ValidationConstants.INVALID_MESSAGE + " (" + dto.getCdsPosition() + ")");
+			} else {
+				cdsStart = cdsStartEnd.getLeft();
+				cdsEnd = cdsStartEnd.getRight();
+			}
+		}
+		predictedVariantConsequence.setCalculatedCdsStart(cdsStart);
+		predictedVariantConsequence.setCalculatedCdsEnd(cdsEnd);
+		
+		Integer proteinStart = null;
+		Integer proteinEnd = null;
+		if (StringUtils.isNotBlank(dto.getProteinPosition())) {
+			Pair<Integer, Integer> proteinStartEnd = parseStartEnd(dto.getProteinPosition());
+			if (proteinStartEnd == null) {
+				response.addErrorMessage("proteinPosition", ValidationConstants.INVALID_MESSAGE + " (" + dto.getProteinPosition() + ")");
+			} else {
+				proteinStart = proteinStartEnd.getLeft();
+				proteinEnd = proteinStartEnd.getRight();
+			}
+		}
+		predictedVariantConsequence.setCalculatedProteinStart(proteinStart);
+		predictedVariantConsequence.setCalculatedProteinEnd(proteinEnd);
+		
+		if (response.hasErrors()) {
+			throw new ObjectValidationException(dto, response.errorMessagesString());
+		}
+		
+		predictedVariantConsequence = predictedVariantConsequenceDAO.persist(predictedVariantConsequence);
+			
+		if (!isUpdate) {
+			if (variantLocation.getPredictedVariantConsequences() == null) {
+				variantLocation.setPredictedVariantConsequences(new ArrayList<>());
+			}
+			variantLocation.getPredictedVariantConsequences().add(predictedVariantConsequence);
+		}
+		cvglaDAO.persist(variantLocation);
+		
+		return predictedVariantConsequence;
+	}
+	
+	private Map<String, String> getExtraAttributes(VepTxtDTO dto) {
+		Map<String, String> attributes = new HashMap<String, String>();
+		if (CollectionUtils.isNotEmpty(dto.getExtra())) {
+			for (String keyValue : dto.getExtra()) {
+				String[] parts = keyValue.split("=");
+				if (parts.length == 2) {
+					attributes.put(parts[0], parts[1]);
+				}
+			}
+		}
+		
+		return attributes;
+	}
+	
+	private Pair<VocabularyTerm, Float> parsePathogenicityPredictionScore(String result, String vocabularyName) {
+		
+		if (StringUtils.isBlank(result)) {
+			return null;
+		}
+		
+		Matcher matcher = PATHOGENICITY_PREDICTION_RESULT.matcher(result);
+		if (!matcher.find()) {
+			return null;
+		}
+		
+		VocabularyTerm consequence = vocabularyTermService.getTermInVocabulary(vocabularyName, matcher.group(1)).getEntity();
+		if (consequence == null) {
+			return null;
+		}
+		
+		ImmutablePair<VocabularyTerm, Float> parsedResult = new ImmutablePair<>(consequence, Float.parseFloat(matcher.group(2)));
+		
+		return parsedResult;
+	}
+	
+	private Pair<Integer, Integer> parseStartEnd(String position) {
+		Matcher matcher = POSITION_STRING.matcher(position);
+		if(!matcher.find()) {
+			return null;
+		}
+		
+		Integer start = null;
+		Integer end = null;
+		String[] positions = position.split("-");
+		
+		if (positions.length > 2) {
+			return null;
+		}
+		
+		if(positions.length == 1) {
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.
<!-- __reviewdog__:ChA2YWZlMGQyYTZkNWIwYWFiEgpjaGVja3N0eWxl -->